azurelinuxagent/common/cgroupapi.py

```diff
@@ -423,14 +427,53 @@ def start_extension_command(self, extension_name, command, shell, cwd, env, stdo
             env=env,
             preexec_fn=os.setsid)
 
-        logger.info("Started extension using scope '{0}'", scope_name)
+        # Wait a bit and check if we completed with error
+        time.sleep(1)
+        return_code = process.poll()
+
+        if return_code is not None and return_code != 0:
+            process_output = read_output(stdout, stderr)
+
+            # When systemd-run successfully invokes a command, thereby creating its unit, it will output the
+            # unit's name. Since the scope name is only known to systemd-run, and not to the extension itself,
+            # if scope_name appears in the output, we are certain systemd-run managed to run.
+            if scope_name not in process_output:
+                logger.warn('Failed to run systemd-run for unit {0}.scope '
+                            'Process exited with code {1} and output {2}'.format(scope_name,
+                                                                                 return_code,
+                                                                                 process_output))
+
+                add_event(AGENT_NAME,
```

Comment:
It would, although it would log it as error instead of warning. There's no need for two identical log messages (with different warning levels), but it means this failure will always be logged as an error.